### PR TITLE
implement repr for untyped lists

### PR DESCRIPTION
**Pull request by esc**
This implemes a proper `__repr__` for untyped numba.typed.List and
tests.

### CODE REVIEW

File: numba/typed/typedlist.py [hunk 1]
```diff
@@ -397,7 +397,7 @@ def __str__(self):
 
     def __repr__(self):
         body = str(self)
-        prefix = str(self._list_type)
+        prefix = str(self._list_type) if self._typed else "ListType[UNTYPED]"
```
Comment (stuartarchibald) [inline numba/typed/typedlist.py:400]: Perhaps a `ListType[Undefined]` ? As in, we know it's a `ListType` but the item type is `Undefined`, I think this makes it more explicit and consistent with other things (which tend to refer to "undefined")?

Comment (esc) [reply numba/typed/typedlist.py]: Fixed

**Comment by esc** [comment]
Thanks for merge.

**Pull request merged**